pkg/kubectl/cmd/create_deployment.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+const (
+	deploymentLong = `
+Create a deployment with the specified name.`
+
+	deploymentExample = `  # Create a new deployment named my-dep
+  kubectl create deployment my-dep --image=busybox`
+)
+
+// NewCmdCreateDeployment is a macro command to create a new deployment
+func NewCmdCreateDeployment(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "deployment NAME --image=image [--dry-run]",
+		Aliases: []string{"dep"},
+		Short:   "Create a deployment with the specified name.",
+		Long:    deploymentLong,
+		Example: deploymentExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateDeployment(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, cmdutil.DeploymentBasicV1Beta1GeneratorName)
+	cmd.Flags().StringSlice("image", []string{}, "Image name to run.")
+	return cmd
+}
+
+// CreateDeployment implements the behavior to run the create deployment command
+func CreateDeployment(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, args []string) error {
```

Comment:
If `--image` is always required, you need to validate that. [Here's an example](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/create_secret.go#L238-L243).